promql/engine.go

```diff
@@ -654,13 +654,10 @@ func (ng *Engine) populateSeries(ctx context.Context, querier storage.Querier, s
 		// The evaluation of the VectorSelector inside then evaluates the given range and unsets
 		// the variable.
 		evalRange time.Duration
-		warnings  storage.Warnings
-		err       error
+		manager   = querier.NewManager()
```

Comment:
As discussed offline I would just wrap querier in some `lazyQuerier struct`

so:

`lazyQuerier := NewLazyQuerier(querier)`

or with some opts `q.concurrency == 1`  we will do:

`lazyQuerier := NewSequentialQuerier(querier)`

No need to touch even storage interfaces